crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -1,12 +1,45 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
 use ruff_python_ast::MatchCase;
 
+use crate::expression::maybe_parenthesize_expression;
+use crate::expression::parentheses::Parenthesize;
+use crate::not_yet_implemented_custom_text;
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
+
 #[derive(Default)]
 pub struct FormatMatchCase;
 
 impl FormatNodeRule<MatchCase> for FormatMatchCase {
     fn fmt_fields(&self, item: &MatchCase, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let MatchCase {
+            range: _,
+            pattern: _,
+            guard,
+            body,
+        } = item;
+
+        write!(
+            f,
+            [
+                text("case"),
+                space(),
+                not_yet_implemented_custom_text("NOT_YET_IMPLEMENTED_Pattern"),
```

Comment:
We can also do this as a separate PR: Ideally, we call into `pattern.format()` here. This requires implementing `AsFormat` and `IntoFormat` for `Pattern`, similar to how it is done for `Mod`

https://github.com/astral-sh/ruff/blob/f45e8645d7c4a2ad4e934ebbbadd91ca59fd543a/crates/ruff_python_formatter/src/module/mod.rs#L1-L35

Reply:
Yes, it's already a work in progress in the next PR (https://github.com/astral-sh/ruff/pull/6360/files#diff-05760c9829e7694c86aaf97014143c0785c12da546f06dcfbc87caf67b15d773) ;)